internal/rbd/nodeserver.go

```diff
@@ -789,6 +789,24 @@ func (ns *NodeServer) NodeExpandVolume(
 	if err != nil {
 		return nil, status.Error(codes.Internal, err.Error())
 	}
+	mapperFile, mapperPath := util.VolumeMapper(volumeID)
+	_, mapper, err := util.DeviceEncryptionStatus(ctx, mapperPath)
+	if err != nil {
+		util.ErrorLog(ctx, "error determining LUKS device on %s (%s): %w",
+			mapperPath, devicePath, err)
+		return &csi.NodeExpandVolumeResponse{}, err
+	}
+	if len(mapper) > 0 {
+		// The volume is encrypted, resize an active mapping
+		err = util.ResizeEncryptedVolume(ctx, mapperFile)
+		if err != nil {
+			util.ErrorLog(ctx, "failed to resize device %s, mapper %s: %w",
+				devicePath, mapperFile, err)
+			return &csi.NodeExpandVolumeResponse{}, err
+		}
+		// Use mapper device path for fs resize
+		devicePath = mapperPath
+	}
```

Comment:
This is required only if the device is encrypted?  We are storing the encryption state in the metadata file, Have an if check here to do this only if it's encrypted.

Reply:
The above block is only executed if the device is encrypted right ? mapper is only set for the encrypted devices.

Or did I missed anything in your question?

Reply:
Yes this works, my idea was to use something like existing https://github.com/ceph/ceph-csi/blob/276918db1080e8e42aca19e2ca02312e6e240c9d/internal/rbd/rbd_attach.go#L333-L351 to avoid running `DeviceEncryptionStatus` for non encrypted devices

Reply:
Maybe you are pointing to a different code block here.
But I understand your intention is to suggest getting the encryption status from the StashFile rather than running some real commands. Sure we can save some jiffies I guess.

Reply:
Encrypted devices have some metadata set, you can run `rbdVolume.checkRbdImageEncrypted()` or similar. Unfortunately there is not `rbdVolume` object created in this function :disappointed:

Reply:
Yeah, let's check the stashFile :+1: